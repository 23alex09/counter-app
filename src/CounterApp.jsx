import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () =>
        setCounter( counter + 1 );


    const handleSubstract = () =>
        setCounter( counter - 1 );


    const handleReset = () => setCounter( 0 );


    return (
        <>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>
            <button aria-label="btn-inc" onClick={ handleAdd }> +1 </button>
            <button aria-label="btn-dec" onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
