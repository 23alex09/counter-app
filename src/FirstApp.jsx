import PropTypes from 'prop-types';

export const FirstApp = ( { title, subtitle, name } ) => {
    //console.log(props)
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <h2>{ subtitle }</h2>
            <p>{ name }</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}

FirstApp.defaultProps = {
    subtitle: 1000,
    name: 'Alejandro'
}