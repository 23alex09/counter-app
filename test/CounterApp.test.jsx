import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe( 'test suit for ConterApp component', () => {

    const initialValue = 0;
    test( 'should match with the snapshot', () => {
        const { container } = render( <CounterApp value={ initialValue } /> )
        expect( container ).toMatchSnapshot();
    } )

    test( 'should show the initial value 0', () => {
        render( <CounterApp value={ 100 } /> )
        expect( screen.getByText( 100 ) ).toBeTruthy();
    } )

    test( 'should increment pressing button +1', () => {
        render( <CounterApp value={ initialValue } /> )
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-inc' } ) );
        expect( screen.getAllByText( initialValue + 1 ) ).toBeTruthy();
    } )

    test( 'should decrement pressing button -1', () => {
        render( <CounterApp value={ initialValue } /> )
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-dec' } ) );
        expect( screen.getAllByText( initialValue - 1 ) ).toBeTruthy();
    } )

    test( 'should show 0 pressing button Reset', () => {
        render( <CounterApp value={ initialValue } /> )
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-inc' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-inc' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-inc' } ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-reset' } ) );
        expect( screen.getAllByText( 0 ) ).toBeTruthy();
    } )
} )