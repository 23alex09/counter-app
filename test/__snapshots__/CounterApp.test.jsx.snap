// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for ConterApp component should match with the snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button
    aria-label="btn-inc"
  >
     +1 
  </button>
  <button
    aria-label="btn-dec"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
